python/flare/idb2pat.py

```diff
@@ -189,134 +231,118 @@ def make_func_sig(config, func):
     """
     logger = logging.getLogger("idb2pat:make_func_sig")
 
-    if func.endEA - func.startEA < config.min_func_length:
+    if func.end_ea - func.start_ea < config.min_func_length:
         logger.debug("Function is too short")
         raise FuncTooShortException()
 
-    ea = func.startEA
-    publics = []  # type: idc.ea_t
+    publics = []  # type: tuple(idc.ea_t, name)
     refs = {}  # type: dict(idc.ea_t, idc.ea_t)
     variable_bytes = set([])  # type: set of idc.ea_t
 
-    while ea != BADADDR and ea < func.endEA:
-        logger.debug("ea: %s", hex(ea))
+    func_len = func.end_ea - func.start_ea
+
+    # get function name and all public names in function
+    func_name = idc.get_func_name(func.start_ea)
+    if func_name == "":
+        func_name = "?" # pat standard
 
-        name = get_name(ea)
-        if name is not None and name != "":
-            logger.debug("has a name")
-            publics.append(ea)
+    ea = func.start_ea
+    while ea != idc.BADADDR and ea < func.end_ea:
+        if ea > func.start_ea:  # skip first byte
+            name = idc.get_name(ea)
+            if name != "" and idaapi.is_uname(name):
+                logger.debug(str("ea 0x%X has a name %s" % (ea, name)))
```

Comment:
first parameter to a log method is a format string, so we can do this instead:

    logger.debug("ea 0x%X has a name %s", ea, name)